### PR TITLE
Remove the assumption that n/(k+1) is a multiple of 8

### LINKED ISSUE #1148: Equihash: remove constraint that n/(k+1) is a multiple of 8
I think this will be necessary to choose parameters that solve the time-to-solution problem.

### PR DESCRIPTION
**Pull request by str4d**
This version works, but generates the initial rows in a way that is probably
not what we want to specify.

Closes #1148

### CODE REVIEW

**Comment by str4d** [comment]
When we are happy with this, I will open a new issue for fixing the implementation to match what we want to specify.

File: src/crypto/equihash.cpp [hunk 1]
```diff
@@ -72,15 +72,30 @@ eh_index UntruncateIndex(const eh_trunc t, const eh_index r, const unsigned int
...
     state = base_state;
     unsigned char array[sizeof(eh_index)];
     eh_index lei = htole32(i);
     memcpy(array, &lei, sizeof(eh_index));
     crypto_generichash_blake2b_update(&state, array, sizeof(eh_index));
-    crypto_generichash_blake2b_final(&state, hash, n/8);
+
+    if (8*cByteLen == cBitLen) {
+        // We are colliding an integer number of bytes; hLen == N/8
+        crypto_generichash_blake2b_final(&state, hash, hLen);
```
Comment (daira) [inline src/crypto/equihash.cpp]: Keeping the hash call outside the ``if`` would probably result in easier-to-follow code.

Comment (daira) [inline src/crypto/equihash.cpp]: Note that even when we change the expansion method, the same expression should work for the hash output length in both cases.

**Comment by daira** [comment]
ACK.

The first commit has me as committer and Signed-off-by even though it's not identical to the code I wrote. In such cases can you recommit as you, and add "Author: Daira Hopwood" in the commit message please?

**Comment by str4d** [comment]
Rebased on .latest and fixed the commit message.

**Comment by daira** [comment]
ACK

**Comment by bitcartel** [comment]
ACK

@zkbot r+

**Pull request merged**